### PR TITLE
Integrate faucet in go-obscuro

### PR DESCRIPTION
**Pull request by otherview**
### Why this change is needed

https://github.com/obscuronet/obscuro-internal/issues/1784

### What changes were made as part of this PR

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

**Comment by moraygrieve** [comment]
The below in manual-deploy-testnet-l2.yml wont work as they do a repository dispatch to the faucet repo ...  
 
  ```
deploy-faucet:
    runs-on: ubuntu-latest
    needs:
      - check-obscuro-is-healthy
    steps:
      - name: 'Trigger Faucet deployment for testnet on a new deployment'
        if: ${{ (github.event.inputs.testnet_type == 'testnet') }}
        run: |
          curl -XPOST -H "Authorization: Bearer ${{ secrets.GH_TOKEN }}" -H "Accept:application/vnd.github" -H "Content-Type:application/json" https://api.github.com/repos/obscuronet/faucet/dispatches --data '{ "event_type": "testnet_deployed", "client_payload": { "ref": "${{ github.ref }}" }'

      - name: 'Trigger Faucet deployment for dev-testnet on a new deployment'
        if: ${{ (github.event.inputs.testnet_type == 'dev-testnet') }}
        run: |
          curl -XPOST -H "Authorization: Bearer ${{ secrets.GH_TOKEN }}" -H "Accept:application/vnd.github" -H "Content-Type:application/json" https://api.github.com/repos/obscuronet/faucet/dispatches --data '{ "event_type": "dev_testnet_deployed", "client_payload": { "ref": "${{ github.ref }}" }'

```

File: tools/faucet/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Obscuro Faucet
+
+This repository contains a Faucet to allow allocation of OBX tokens within an Obscuro network. For more information
```
Comment (moraygrieve) [inline tools/faucet/README.md:3]: `This repository` is no longer valid

File: tools/faucet/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Obscuro Faucet
+
+This repository contains a Faucet to allow allocation of OBX tokens within an Obscuro network. For more information 
+on Obscuro see the [Obscuro repo](https://github.com/obscuronet/go-obscuro) and [documentation](https://docs.obscu.ro/).
+
+## Repository Structure
```
Comment (moraygrieve) [inline tools/faucet/README.md:6]: Same here ... the readme needs updating after the move

**Comment by moraygrieve** [comment]
Any reason why building of the faucet is not included in `testnet-local-build_images.sh ` e.g. obscuroscan is a tool and the docker image is built in that script?

File: .github/workflows/manual-deploy-testnet-faucet.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        type: choice
+        options:
+          - 'dev-testnet'
+          - 'testnet'
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (moraygrieve) [inline .github/workflows/manual-deploy-testnet-faucet.yml:29]: Do you want to update this to @v3 like you have done elsewhere?

File: tools/faucet/README.md (hunk 1 above)
Comment (otherview) [reply tools/faucet/README.md:3]: Fixed 👍

**Comment by otherview** [comment]
> Any reason why building of the faucet is not included in `testnet-local-build_images.sh ` e.g. obscuroscan is a tool and the docker image is built in that script?

Added 👍

**Comment by moraygrieve** [comment]
Note that the top level readme (go-obscuro/README.md) will need to be updated also as it contains a section on starting up the faucet referencing it from the faucet repo

**Review by moraygrieve — APPROVED** [review]
LGTM

**Pull request merged**